### ISSUE TITLE
[3.3.5][6.x] Core/Spell: Val'anyr, Hammer of Ancient Kings - Procc

### ISSUE THREAD

**Issue by Kittnz** [TrinityCore/TrinityCore#16478]
[Val'anyr, Hammer of Ancient Kings] does proc on the target that you heal, however if you are a paladin and if you use  [Glyph of Holy Light] the proc of the weapon does not get apply'd the 5 other friendly targets.

Info:
http://wotlk.openwow.com/item=41106
http://wotlk.openwow.com/item=46017

http://wotlk.openwow.com/spell=64413

0813261c6fff568de2a3ce8df4b93c77d2e10d21

**Comment by Keader** [comment]
Healing Over Time shold proc?

**Comment by Kittnz** [comment]
the glyph works but the proc of valanyr proc (the shield) does not proc on the people who got the glyph healing

**Comment by Keader** [comment]
yes... but healing over time should proc valanyr?
Because renew (priest) proc her

**Comment by Kittnz** [comment]
yes it should, every healing ability

**Comment by Keader** [comment]
It's strange, because a few months ago that did not happen. Moreover, why Rejuvenation (Druid Resto Spell) not active?

**Comment by jackpoz** [comment]
"Priority-Critical" category is for crashes that happen every hour to a broad range of users or databases getting corrupted daily

**Comment by Keader** [comment]
According wow head, glyph shold proc

**Comment by Kittnz** [comment]
It should, and it isn't =)

@jackpoz alright!

**Comment by Keader** [comment]
about beacon of light
On wow head comments:

>Once procd, a paladin's beacon would benefit a ton from this, as both the MT and the OT could get a pretty nice shield.

http://www.wowhead.com/forums&topic=94136/valanyr-hammer-of-the-ancient-kings-proc-info#p1045098

**Comment by Treeston** [comment]
Did some investigating. Manually using .cast to cast [spell #54968](http://wotlk.openwow.com/spell=54968) doesn't proc Val'anyr's shield either.

**Comment by ariel-** [comment]
It's because spell 54968 Glyph of Holy Light has DAMAGE_CLASS_NONE, and 64411 Blessing of Ancient Kings procs only of spells with DAMAGE_CLASS_MAGIC

fix:
```sql
DELETE FROM `spell_proc_event` WHERE `entry`=64411;
INSERT INTO `spell_proc_event` (`entry`,`SchoolMask`,`SpellFamilyName`,`SpellFamilyMask0`,`SpellFamilyMask1`,`SpellFamilyMask2`,`procFlags`,`procEx`,`ppmRate`,`CustomChance`,`Cooldown`) VALUES
(64411,0,0,0,0,0,0x00044400,0,0,0,0);
```

**Comment by Treeston** [comment]
Shouldn't we instead use DAMAGE_CLASS_MAGIC for Glyph of HL?

**Comment by ariel-** [comment]
I don't know, Earth shield is another heal with DAMAGE_CLASS_NONE, so perhaps it has some meaning in blizzard, or maybe procflag 0x4000 should proc off DMG_CLASS_NONE heals, too

**Comment by Treeston** [comment]
Does/did earth shield proc val'anyr?

**Comment by ariel-** [comment]
If I understand correctly the procflags, it won't proc without the above sql patch, but ingame testing just to be 100% sure

**Comment by Treeston** [comment]
Oh, the sql changes the proc mask. I see.

**Comment by ariel-** [comment]
As previously stated, earth shield triggered heal does not proc off the Protection of Ancient Kings shield, when modifying the procFlag to include dmgclass none spells, it does

**Comment by Keader** [comment]
Something break it again in Rev 971ed856a49dc56ea33d9a0779b1386eea5cd3ca
Is not working with Glyph of Holy Light.
Add PROC_ATTR_TRIGGERED_CAN_PROC will make proc, but will break proc with Judgement of Light (should not proc with Judgement)

**Comment by Keader** [comment]
@ariel-  / @ccrs  any tips?

**Comment by Treeston** [comment]
Figure out which exact rev breaks it, it's definitely not 971ed85.

**Comment by Keader** [comment]
Nope, i mean in rev 971ed856a49dc56ea33d9a0779b1386eea5cd3ca (last rev) is not working again xD

**Comment by Treeston** [comment]
> Figure out which exact rev breaks it

otherwise "any tips" will be hard to provide

**Comment by Keader** [comment]
hmm, looks like that spell_proc rewrite broken it again.
In a very old core, before proc rewrite was working fine.

**Comment by Jildor** [comment]
Fix:
```sql
UPDATE spell_proc SET `AttributesMask`=2 WHERE `SpellID`=64411;
```

The problem with Judgement of Light is not related with the Fix, I think is related with:
```AttributesEx3: 0x04000000 (SPELL_ATTR3_CAN_PROC_WITH_TRIGGERED)```
of spell: ID - 64415 Val'anyr Hammer of Ancient Kings - Equip Effect

**Comment by Keader** [comment]
add PROC_ATTR_TRIGGERED_CAN_PROC will make val'anyr proc with trinket's heal, like Glowing (Halion trinket) and not should proc with glowing.

**Comment by Jildor** [comment]
oh, I had not noticed...
